File: answer.pseudo.js
//Returns name with a random number between min and max
function generator(name, min, max){

    name = name || 'Mansoor';
    min = min || 6;
    max = max || 15;

    //generate a random number between max and min, both exclusive
    var rndNumber = Math.ceil(Math.random() * (max - min - 1)) + min;

    //concatenate name with random number and return it
    return name.concat(" ", rndNumber);
}

console.log(generator());

File: answer.pseudo.rb
def generator(name="Mansoor", min="6", max="15")

  # modify min and max so the random number does not
  # include min and max
  min = min.to_i + 1
  max = max.to_i - 1

  # concatenate name with random number and return it
  name + " " + rand(min..max).to_s

end

print generator